lib/pacemaker/pcmk_sched_allocate.c

```diff
@@ -151,7 +151,8 @@ check_failure_threshold(pe_resource_t *rsc, pe_node_t *node)
 static void
 apply_exclusive_discovery(pe_resource_t *rsc, pe_node_t *node)
 {
-    if (rsc->exclusive_discover || uber_parent(rsc)->exclusive_discover) {
+    if (rsc->exclusive_discover
+        || pe__const_top_resource(rsc, false)->exclusive_discover) {
```

Comment:
That does make sense. I suspect it's not relevant in practice because resource discovery is set via location constraints, and constraints are always supposed to reference the topmost resource. The various usages of ->exclusive_discover do seem a bit inconsistent though.

Reply:
Yeah, bundles are convoluted, but I think this is special-cased for them. Also, resource discovery applies only to location constraints, not colocations.

Bundle variant data has a child member, which is a clone of the inner primitive (if any). The data also has a list of replicas, which are structs that also have a child member, and each replica's child is set to one of the clone instances. The instance's own ->exclusive_discover is forced to TRUE at that time (since each instance is bound to the replica's guest node). So here, the rsc->exclusive_discover is what matters.

Reply:
> Also, resource discovery applies only to location constraints, not colocations.

Yeah I wasn't clear, I didn't mean **this** specifically (exclusive_discover) but rather this general concept of not checking a parameter in (or not applying a parameter to) everything in the hierarchy.

Reply:
Ah yes, I do suspect there are issues lurking around that